waspc/data/Generator/templates/server/src/jobs/core/pgBoss/pgBossJob.js

```diff
@@ -0,0 +1,84 @@
+import { boss } from './pgBoss.js'
+import { Job } from '../Job.js'
+import { SubmittedJob } from '../SubmittedJob.js'
+
+export const PG_BOSS_EXECUTOR_NAME = Symbol('PgBoss')
+
+/**
+ * A PgBoss specific SubmittedJob that adds additional PgBoss functionality.
+ */
+class PgBossSubmittedJob extends SubmittedJob {
+  constructor(job, jobId) {
+    super(job, jobId)
+    this.pgBoss = {
+      async cancel() { return boss.cancel(jobId) },
+      async resume() { return boss.resume(jobId) },
+      async details() { return boss.getJobById(jobId) }
```

Comment:
I'm guessing `boss.something(jobId)` methods all return promises. If that's the case, `async` is redundant.

Also, I think these functions are simple enough to use arrow functions (if you want):
```javascript
this.pgBoss = {
  cancel: () => boss.cancel(jobId),
  resume: () => boss.resume(jobId),
  details: () => boss.getJobById(jobId),
}
```

Reply:
Good catch and suggestion, thanks 👍🏻